cel/cel_test.go

```diff
@@ -1857,3 +1858,110 @@ func TestDeclareContextProto(t *testing.T) {
 		t.Fatalf("env.Compile(%s) failed: %s", expression, iss.Err())
 	}
 }
+
+type testRegexCall struct {
+	interpreter.InterpretableCall
+	compiledRegex *regexp.Regexp
+}
+
+func (r testRegexCall) Eval(ctx interpreter.Activation) ref.Val {
+	args := r.InterpretableCall.Args()
+	argVals := make([]ref.Val, len(args))
+	for i, arg := range args {
+		argVals[i] = arg.Eval(ctx)
+		if types.IsUnknownOrError(argVals[i]) {
+			return argVals[i]
+		}
+	}
+	return types.Bool(r.compiledRegex.MatchString(argVals[0].Value().(string)))
+}
+
+var testRegexOptimization = &interpreter.RegexOptimization{
+	Function:   "matches",
+	RegexIndex: 1,
+	Factory: func(call interpreter.InterpretableCall, regexIndex int, pattern ref.Val) (interpreter.Interpretable, error) {
+		compiledRegex, err := regexp.Compile(pattern.Value().(string))
+		if err != nil {
+			return nil, err
+		}
+		return &testRegexCall{call, compiledRegex}, nil
+	},
+}
+
+func TestRegexOptimizer(t *testing.T) {
```

Comment:
Consider adding a test to the `interpreter/interpreter_test.go` with a default implementation for `matches` that's enabled when the `OptOptimize` flag is set.

Reply:
Sounds good. I've tried to fix this together in a reasonable way.